kernel/boot_info/src/uefi.rs

```diff
@@ -0,0 +1,242 @@
+    type MemoryRegion<'a> = MemoryRegion;
+    type MemoryRegions<'a> = MemoryRegions;
+
+    type ElfSection<'a> = &'a info::ElfSection;
+    type ElfSections<'a> = core::slice::Iter<'a, info::ElfSection>;
+
+    type Module<'a> = Module;
+    type Modules<'a> = Modules;
+
+    fn start(&self) -> Option<VirtualAddress> {
+        VirtualAddress::new(*self as *const _ as usize)
+    }
+
+    fn len(&self) -> usize {
+        self.size
+    }
+
+    // The bootloader creates two memory regions with the bootloader type. The first
+    // one always starts at 0x1000 and contains the page table, boot info, etc. The
+    // second one starts at some other address and contains the nano_core elf file.
+    // It is the same size as the nano_core elf file.
+
+    fn kernel_memory_range(&self) -> Result<Range<PhysicalAddress>, &'static str> {
+        use crate::ElfSection;
+
+        let start = PhysicalAddress::new(
+            self.elf_sections()?
+                .into_iter()
+                .filter(|s| s.flags().contains(ElfSectionFlags::ALLOCATED))
+                .map(|s| s.start())
+                .min()
+                .ok_or("couldn't find kernel start address")?
+                .value(),
+        )
+        .ok_or("kernel physical start address was invalid")?;
+        let virtual_end = self
+            .elf_sections()?
+            .into_iter()
+            .filter(|s| s.flags().contains(ElfSectionFlags::ALLOCATED))
+            .map(|s| s.start() + s.len())
+            .max()
+            .ok_or("couldn't find kernel end address")?;
+        let physical_end = PhysicalAddress::new(virtual_end.value() - KERNEL_OFFSET)
+            .ok_or("kernel physical end address was invalid")?;
+
+        Ok(start..physical_end)
+    }
+
+    fn bootloader_info_memory_range(&self) -> Result<Range<PhysicalAddress>, &'static str> {
+        let mut iter = self
+            .memory_regions
+            .iter()
+            .filter(|region| region.kind == info::MemoryRegionKind::Bootloader)
+            .filter(|region| region.start == 0x1000);
+
+        let bootloader_info_memory_region =
+            iter.next().ok_or("no bootloader info memory region")?;
+        if iter.next().is_some() {
+            Err("multiple potential bootloader memory info memory regions")
+        } else {
+            let start = PhysicalAddress::new(bootloader_info_memory_region.start as usize)
+                .ok_or("invalid bootloader info start address")?;
+            let end = PhysicalAddress::new(bootloader_info_memory_region.end as usize)
+                .ok_or("invalid bootloader info end address")?;
+            Ok(start..end)
+        }
+    }
+
+    fn modules_memory_range(&self) -> Result<Range<PhysicalAddress>, &'static str> {
+        let area = self
+            .memory_regions
+            .iter()
+            .find(|region| region.kind == info::MemoryRegionKind::UnknownUefi(0x80000000))
```

Comment:
same comment here, use something like 
```rust
const UefiModuleMemoryKind: ... = info::MemoryRegionKind::UnknownUefi(0x80000000);
```
or whatever you feel is the best name for it